### PR TITLE
Add ISB003 analyzer to validate exported brokered services

### PR DESCRIPTION
**Pull request by AArnott**
Fixes devdiv-1914458 by calling out the mistake at compile-time.

### CODE REVIEW

File: src/Microsoft.ServiceHub.Analyzers/Strings.resx [hunk 1]
```diff
@@ -147,4 +147,14 @@
   <data name="ISB002_Title" xml:space="preserve">
     <value>Avoid storing rentals in fields</value>
   </data>
+  <data name="ISB003_Description" xml:space="preserve">
+    <value>Classes attributed with ExportBrokeredServiceAttribute must implement  IExportedBrokeredService.</value>
```
Comment (RyanToth3) [inline src/Microsoft.ServiceHub.Analyzers/Strings.resx:151]: Remove extra space

File: test/Microsoft.ServiceHub.Analyzers.Tests/Microsoft.ServiceHub.Analyzers.Tests.csproj [hunk 2]
```diff
@@ -21,4 +21,8 @@
     <ProjectReference Include="..\..\src\Microsoft.ServiceHub.Framework\Microsoft.ServiceHub.Framework.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Using Include="Xunit" />
```
Comment (RyanToth3) [inline test/Microsoft.ServiceHub.Analyzers.Tests/Microsoft.ServiceHub.Analyzers.Tests.csproj:25]: Does this add an implicit using to all files in the project?

**Comment by RyanToth3** [comment]
Should there also be a Warning for if someone implements the interface but forgets to add the attribute?

File: test/Microsoft.ServiceHub.Analyzers.Tests/Microsoft.ServiceHub.Analyzers.Tests.csproj (hunk 2 above)
Comment (AArnott) [reply test/Microsoft.ServiceHub.Analyzers.Tests/Microsoft.ServiceHub.Analyzers.Tests.csproj:25]: Yes.

**Comment by AArnott** [comment]
> Should there also be a Warning for if someone implements the interface but forgets to add the attribute?

An interesting question. It hasn't come up yet. And as you say, this feels more like a _warning_ than the _error_ case, which means I need a new diagnostic ID. We needed an analyzer for this because an attribute without the interface breaks VS. But an interface without the attribute doesn't break VS. I also suspect it won't be as common a user mistake because exporting a class almost always requires an export attribute.... but an export attribute that requires a specific interface to be implemented (and particularly an interface that _isn't_ even specified in the export attribute) is far more an unusual requirement, thus leading at least one partner team to miss it.

File: test/Microsoft.ServiceHub.Analyzers.Tests/Microsoft.ServiceHub.Analyzers.Tests.csproj (hunk 2 above)
Comment (RyanToth3) [reply test/Microsoft.ServiceHub.Analyzers.Tests/Microsoft.ServiceHub.Analyzers.Tests.csproj:25]: Very cool

**Review by RyanToth3 — APPROVED** [review]

**Pull request merged**